File: 백준/Silver/14425. 문자열 집합/문자열 집합.py
import sys

def main() -> None:
    input = sys.stdin.readline
    n, m = map(int, input().split())
    s = {input().strip() for _ in range(n)}
    # n + m 줄의 입력이 들어오ㅓ고 n개의 문자열은 그룹 s, 그 외 나머지는 검사 그룹
    print(sum(1 for _ in range(m) if input().strip() in s))

if __name__ == "__main__":
    main()

File: 백준/Silver/14425. 문자열 집합/문자열 집합.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((s) => s.trim());

const [N, M] = input[0].split(" ").map(Number);

const n = new Set(input.slice(1, N + 1));

const result = input.slice(N + 1).filter((x) => n.has(x));

console.log(result.length);